File: 1929-concatenation-of-array/1929-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] arr = new int[nums.length * 2];
        for(int i = 0; i < nums.length * 2; i++) {
            if(i >= nums.length) arr[i] = nums[i - nums.length];
            else arr[i] = nums[i];
        }
        return arr;
    }
}

File: 1929-concatenation-of-array/1929-concatenation-of-array.swift
class Solution {
    func getConcatenation(_ nums: [Int]) -> [Int] {
        var res = [Int].init(repeating: 0, count: nums.count * 2)
        var i = 0
        
        while i < nums.count * 2 {
            if i >= nums.count {
                res[i] = nums[i % nums.count]
            } else {
                res[i] = nums[i]
            }
            i += 1
        }
        return res
    }
}